silicon: add page (#8392)
tldr-pages/tldr: pages/common/silicon.md

# silicon

> > Create images of source code files.

Reviewer: Nit: since we can create image for anything, should we drop the `files` ?
```suggestion
> Create images of source code.
```

> > Create beautiful image of your source code.

Reviewer 2: We prefer objective descriptions here
```suggestion
> Create images of source code files.
```

> > More information: <https://github.com/Aloxaf/silicon/>.

Reviewer 2: ```suggestion
> More information: <https://github.com/Aloxaf/silicon>.
```

> - Generate an image from a specific source file:

Reviewer 3: ```suggestion
- Generate the image from a specific source file:
```

There is just one outcome for a specific source file. We can use `the` here because it's a specific outcome for some code.

Reviewer 2: I think this should be `an`, not `the`. cc @CleanMachine1 @sbrl

Reviewer 4: yes, it should be

Reviewer 5: Agreed. `an` rather than `the` sounds good to me?

To add some clarity here, `the` is not grammatically incorrect here - it's just that `an` makes more *semantic* sense, because English is a weird language.

Reviewer 3: When we should ignore grammar rules and prefer a semantic sense? :) I wanna know to leave correct suggestions.

> - Generate image from source file:

Reviewer 2: ```suggestion
- Generate an image from a specific source file:
```

`silicon  {{path/to/code.py}} -o {{path/to/image.png}}`

> - Generate image from source file with special program language syntax highlighting:

Reviewer 2: ```suggestion
- Generate an image from a source file with a specific programing language syntax highlighting (e.g. `rust`, `py`, `js`, etc.):
```

> - Generate an image from a source file with a specific programing language syntax highlighting (e.g. `rust`, `py`, `js`, etc.):

Reviewer 3: ```suggestion
- Generate the image from a source file with a specific programing language syntax highlighting (e.g. `rust`, `py`, `js`, etc.):
```

> `silicon  {{path/to/code.py}} -o {{path/to/image.png}} -l {{program lanauge}}`

Reviewer 2: There is no list of languages but it could be `javascript` or `js`.
```suggestion
`silicon  {{path/to/source_file}} --output {{path/to/output_image.png}} --language {{language_or_file_extension}}`
```

> - Generate image from pipe:

Reviewer 4: ```suggestion
- Generate an image from stdin:
```

> `{{command}} | silicon -o {{path/to/image.png}}`

Reviewer 4: ```suggestion
`{{command}} | silicon --output {{path/to/output_image.png}}`
```

> `silicon  {{path/to/source_file}} --output {{path/to/output_image.png}} --language {{language_or_file_extension}}`

Reviewer 3: ```suggestion
`silicon  {{path/to/source_file}} --output {{path/to/output_image.png}} --language {{language|extension}}`
```

> - Generate an image from stdin:

Reviewer 3: ```suggestion
- Generate the image from `stdin`:
```
